paper-api/src/main/java/io/papermc/paper/math/Rotation.java

```diff
@@ -0,0 +1,23 @@
+package io.papermc.paper.math;
+
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Represents a rotation with specified pitch and yaw values.
+ */
+@NullMarked
+public interface Rotation {
```

Comment:
There should be some way to obtain an instance of this if we want it to be useful outside of commands, RotationImpl can be moved into the math package in api and a static method can be added in this class, the same is currently done in Position for example.